packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -446,6 +459,17 @@ export class GarbageCollector implements IGarbageCollector {
             && !gcOptions.disableGC
         );
 
+        const currentVersion = pkgVersion;
+        const oldestTrackGCBlobStateVersion = this.mc.config.getString(trackGCBlobStateVersionKey);
+        const versionIsCompatible =
+            oldestTrackGCBlobStateVersion === undefined ||
+            currentVersionGreaterOrEqualToVersion(currentVersion, oldestTrackGCBlobStateVersion);
+        const trackGCBlobState = this.mc.config.getBoolean(trackGCBlobStateKey) === true;
+        this.trackGCBlobState =
+            this.shouldRunGC &&
```

Comment:
I don't think this needs to check for shouldRunGC. None of the code that uses or updates this will run if GC is not enabled.

Reply:
I was thinking it was better to maintain state than to just assume shouldRunGC to be true.

Reply:
Either way is fine. I was thinking more from a separation of concern perspective. This feature (tracking state) shouldn't care about GC enabled or not. It does its thing and lets the overall GC worry about using it or not.
Compare that with features such as session expiry which should not even run if GC is disabled.

Reply:
Ok, I'll remove this.shouldRunGC